### DISCUSSION TITLE
Possible Features

### DISCUSSION THREAD

**Discussion by DylanVann** [DylanVann/react-native-fast-image#825] [Announcements]
These are major features people are interested in.

-   [ ] Fine grained cache control.
    -   All these must be promise based, work the same across platforms.
    -   Not sure about naming all these yet.
    -   [ ] `getImage`
    -   [ ] `loadImage`
    -   [ ] `setImage`
    -   [ ] `removeImage`
    -   [x] `clearMemoryCache`
    -   [x] `clearDiskCache`
-   [ ] Tint
    -   Needs to work well and look the same across platforms.
-   [ ] Blur
    -   Needs to work well and look the **SAME** across platforms.
-   [x] Fix rounded corner AA issue.
    -   Needs to work per corner, across platforms (should not affect iOS).
    -   Seems to be fixed: https://github.com/DylanVann/react-native-fast-image/pull/411#issuecomment-920326071
-   Probably after Blur and Tint we don't need to worry about arbitrary filters, can be done using other libs.
-   [ ] Image placeholder/defaultSource.
-   [ ] React Native Web support.
    -   Want to at least support all styling features.
    -   Stuff like fine grained cache control and preloading can probably no-op, at least as a first pass.

**Comment by DylanVann** [comment]
There are PRs open for some of these, some of those PRs only partially meet requirements, or they did last time I checked

I'll be trying to merge things related to these features as well as upgrade/chore PRs over the next few days.

**Comment by tomskopek** [comment]
@DylanVann

It seems like a lot of people (myself included) are really keen on have .preload() be promise based or return a callback. This is totally essential for my business logic.

* See this very popular PR that was opened in 2019: https://github.com/DylanVann/react-native-fast-image/pull/437. Several people tried to keep it up to date over the years. It currently has merge conflicts, but I doubt anyone will try to keep it up to date since it seems like a lost cause/ignored PR.
* [This issue](https://github.com/DylanVann/react-native-fast-image/issues/747) raised by @jamesxabregas explains the need for this feature very well
* Another fairly popular [issue](https://github.com/DylanVann/react-native-fast-image/issues/438)

It would be great to know if you intend to implement this feature?
Thanks
